modules/shared/src/main/scala/io/constellationnetwork/security/Hash.scala

```diff
@@ -24,12 +25,38 @@ object hash {
   }
 
   object Hash {
+    private val hexDigits = "0123456789abcdef".toCharArray
+    private val sha256 = MessageDigest.getInstance("SHA-256")
 
+    case class Sha256Digest(private val bytes: Array[Byte]) {
+      require(bytes.length == 32, "invalid SHA-256 digest length")
+
+      def toByteArray: Array[Byte] = Array.copyOf(bytes, bytes.length)
+
+      def toIndexedSeq: IndexedSeq[Byte] = bytes.toIndexedSeq
+
+      def toHexString: String =
+        bytes
+          .foldLeft(new StringBuilder(bytes.length * 2)) { (sb, b) =>
+            sb.append(hexDigits((b >> 4) & 0xf)).append(hexDigits(b & 0xf))
+          }
+          .toString
+
+      override def toString: String = s"Sha256Digest($toHexString)"
```

Comment:
Please implement a Show type class instead of this toString

Reply:
Done

Reply:
there is still the overriding `toString`